### DISCUSSION TITLE
Ability to skip running project's `prepare` script when running `npm install` in the project

### DISCUSSION THREAD

**Discussion by trusktr** [npm/rfcs#261] [RRFC]
I put `prepare` scripts in my projects so that if they are installed as git dependencies in other projects, then `prepare` will build things as needed.

However, when I run `npm install` in a project, I want `prepare` to run only for dependencies that I've installed into my project, and not for the project itself.

Is this possible?

If not, I maybe it would be good feature to add. A good reason for adding such a feature is that running `npm install` only for it to fail because the project is in a broken state is annoying. Sometimes, you just need to install things in the process of getting the project working again.

We can open `package.json` and temporarily remove the `prepare` script as a workaround. It would be nicer though, if I could stick something in my project's `.npmrc` and never have to do that again (only when running `npm install` directly in the project, but the `prepare` script should still always run when it is installed as a git dependency for example).

**Comment by ljharb** [comment]
The best practice is to build prepublish, and not to inflict a build step on every consumer of your package. When would you want `npm install` to work but `npm run prepare` fails? (iow, i don't understand when "sometimes" applies)

**Reply by trajano** [reply]
I am hitting a similar issue, in my case I just want to do an `npm update` of one single package which I know will cause a build error due to the typecheck executed in `prepare` is there a way to get around that?  `--ignore-scripts` do not appear to work.
